### ISSUE TITLE
Move `predict` to StatsAPI.jl

### ISSUE THREAD

**Issue by rafaqz** [JuliaStats/StatsBase.jl#726]
I would like to support `predict` in GeoData.jl for e.g. predicting SDM models over stacks of raster layers, without the round trip to `DataFrame` and back. 

But StatsBase is pretty heavy for a base package, withe nearly a second load time, and GeoData.jl is only loosely related to stats models.

Would it be acceptable to move `predict` to StatsAPI.jl with StatsBase keeping the fallback definition?

**Comment by nalimilan** [comment]
Yes, actually we should move all empty definitions in statsmodel.jl to StatsAPI. Feel free to make a PR.

**Comment by rafaqz** [comment]
By "all empty definitions" do you mean that e.g.

```julia
coef(model::StatisticalModel) = error("coef is not defined for $(typeof(model)).")
```

Would also move to StatsAPI as:

```julia
coef(model) = error("coef is not defined for $(typeof(model)).")
```

**Comment by nalimilan** [comment]
I'd move `coef(model::StatisticalModel)` without changing its signature. Any reason to change it?

**Comment by rafaqz** [comment]
Because the types are defined here ;)

We can move the them too, of course. I just have no idea about the design of these packages.

**Comment by nalimilan** [comment]
Yes we should move them too. There's no reason to force a package that wants to use these types to depend on StatsBase.

**Comment by rafaqz** [comment]
This is done, would it be possible to review the changes over at StatsAPI?

When that is approved/merged I'll make the PR to remove the signatures here and import them from StatsAPI.
